### PR TITLE
chore(tooling): conventional changelog integration

### PR DESCRIPTION
**Pull request by bobylito**
This is a first step towards a better integration of the release step. With that we'll have a way to update the changelog automatically :)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,9 @@
     "test-node": "clear && node test/run",
     "test:watch": "onchange '{src,test}/**/*.js' 'index.js' -- npm run test-node",
     "test:perf": "node ./test/perf/index.js",
-    "release": "git clean dist/ -f && git checkout dist/ && npm install && ./scripts/release.js && npm run build"
+    "release": "git clean dist/ -f && git checkout dist/ && npm install && ./scripts/release.js && npm run build",
```
Comment (vvo) [inline package.json]: where's release.js?

File: scripts/conventional-changelog/index.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+'use strict';
+var parserOpts = {
```
Comment (vvo) [inline scripts/conventional-changelog/index.js]: I never had to pass a script to conventional-changelog, what are the benefits here from the regular conventional-changelog usage?

File: package.json (hunk 1 above)
Comment (bobylito) [inline package.json]: This was not touched, I just added a `,`

File: scripts/conventional-changelog/index.js (hunk 2 above)
Comment (bobylito) [inline scripts/conventional-changelog/index.js]: Seamless integration with 2 years of changelog.

Comment (vvo) [inline scripts/conventional-changelog/index.js]: Oh.. So you are reading back the commit history and generating the changelog out of it somehow?

On some projects what I did is that if the CHANGELOG + state of the project is mixed of good commits and bad commits then I'll configure conventional-changelog to read the **new** commits well and only append to the CHANGELOG file (that I'll rename to CHANGELOG.md). And that's it, we can keep this script but I was a bit surprised to see it (seemed redundant with conventional-changelog defaults)

**Pull request merged**